addons/account/tests/test_account_move_reconcile.py

```diff
@@ -1942,6 +1942,121 @@ def test_reconcile_cash_basis_exchange_difference_transfer_account_check_entries
+                }),
+                (0, 0, {
+                    'debit': 200.0,
+                    'credit': 0.0,
+                    'amount_currency': 400.0,
+                    'currency_id': currency_id,
+                    'account_id': self.company_data['default_account_revenue'].id,
+                }),
+            ]
+        })
+
+        (cash_basis_move + payment_move).action_post()
+
+        self.assertAmountsGroupByAccount([
+            # Account                               Balance     Amount Currency
+            (cash_basis_transition_account,      -33.33,     -100.0),
+            (self.tax_account_1,                    0.0,        0.0),
+        ])
+
+        receivable_lines = (cash_basis_move + payment_move).line_ids\
+            .filtered(lambda line: line.account_id == self.extra_receivable_account_1)
+        res = receivable_lines.reconcile()
+
+        self.assertEqual(len(res.get('tax_cash_basis_moves', [])), 1)
+
+        # Tax values based on payment
+        # Invoice amount 300 (amount currency) with payment rate 2 (400 payment amount divided by 200 invoice balance)
+        #  - Base amount: 150 company currency
+        #  - Tax amount: 50 company currency
+        self.assertRecordValues(res['tax_cash_basis_moves'].line_ids, [
+            # Base amount:
+            {'debit': 150.0,    'credit': 0.0,      'amount_currency': 300.0,   'currency_id': currency_id,     'account_id': self.cash_basis_base_account.id},
+            {'debit': 0.0,      'credit': 150.0,    'amount_currency': -300.0,  'currency_id': currency_id,     'account_id': self.cash_basis_base_account.id},
+            # tax:
+            {'debit': 50.0,     'credit': 0.0,      'amount_currency': 100.0,   'currency_id': currency_id,     'account_id': cash_basis_transition_account.id},
+            {'debit': 0.0,      'credit': 50.0,     'amount_currency': -100.0,  'currency_id': currency_id,     'account_id': self.tax_account_1.id},
+        ])
+
+        exchange_diff = res['full_reconcile'].exchange_move_id
+
+        # Exchange difference
+        # 66.67 amount residual on the payment line after reconciling receivable line of the cash basis move with the payment counterpart
+        # 50.00 difference of the cash_basis_move base line and the CABA entry created by the system
+        self.assertRecordValues(exchange_diff.line_ids, [
+            {'debit': 66.67,    'credit': 0.0,      'currency_id': currency_id,     'account_id': self.extra_receivable_account_1.id},
+            {'debit': 0.0,      'credit': 66.67,    'currency_id': currency_id,     'account_id': self.company_data['company'].income_currency_exchange_account_id.id},
+            {'debit': 50.0,     'credit': 0.0,      'currency_id': currency_id,     'account_id': self.cash_basis_base_account.id},
+            {'debit': 0.0,      'credit': 50.0,     'currency_id': currency_id,     'account_id': self.cash_basis_base_account.id},
```

Comment:
Two times the cash basis base account ? Why ? Those two lines basically cancel each other. They are useful in the cash basis move (to make the amount tax exigible, and hence have it appear in the report); but what is the point of having them in the exchange move ?

I was expecting to see something about cash_basis_transition_account being reconciled with an exchange difference (probably not in this exchange move). Or did I misunderstand something ?

Reply:
> what is the point of having them in the exchange move ?

Well, it is detected as account to fix...
https://github.com/odoo/odoo/blob/03c028811535c18e49de33858da33ad3355484cf/addons/account/models/account_move.py#L4626
...and used twice
https://github.com/odoo/odoo/blob/03c028811535c18e49de33858da33ad3355484cf/addons/account/models/account_move.py#L4712-L4729

> I was expecting to see something about cash_basis_transition_account being reconciled with an exchange difference (probably not in this exchange move). Or did I misunderstand something ?

Indeed those entries are in a separate exchange move, created here
```
1.  reconcile()
    |
2.   -> _create_tax_cash_basis_moves()
         |
3.        -> reconcile()
               |
4.              ->_create_exchange_difference_move() <--------------- HERE
    |
5.   -> _create_exchange_difference_move() 
```

Which generate the lines marked in green in the below image
![Image 1](https://user-images.githubusercontent.com/53174659/141788069-8b057698-db0c-4f85-93f0-7ceafa46bef4.jpg)

The account 209.01.01 in l10n_mx is used as cash_basis_transition_account
![screenshot_018](https://user-images.githubusercontent.com/53174659/143474651-f4fa7429-20f5-4328-aac6-3576e574e008.png)

Reply:
Ok, I got it. The double cash basis base line on the exchange difference move is just done so that its total balance matches the initial base amount. 100 (1/3 of 300) on the invoice become 150 on the cash basis move (1/2 of 300), so we have to do -50 on each repeated base line of the cash basis move (so 2 times, one debit, one credit) to bring everything to 100. Makes sense.